src/protocol/properties.rs

```diff
@@ -214,18 +214,17 @@ pub struct CompletionOptions {
     #[serde(rename = "resolveProvider")]
     pub resolve_provider: Option<bool>,
     #[serde(rename = "triggerCharacters")]
-    pub trigger_characters: Option<Vec<String>>,
+    pub trigger_characters: Option<Vec<char>>,
+    // changed to char due to specs,
```

Comment:
which spec, the LSP spec?

Reply:
Its a string of length 1 in the spec, I would prefer that we keep this as a String to keep the string methods on it for future use, it is the responsibility of the client not to send strings longer than one. If we change this to char and a client sends a string longer than 1 then serialization will fail and the LSP will error as opposed to gracefully not returning any results.